### PR TITLE
Improve jerry_is_feature_enabled with object availability information

### PR DESCRIPTION
**Pull request by ZsoltRaduska**
JerryScript-DCO-1.0-Signed-off-by: Zsolt Raduska rzsolt@inf.u-szeged.hu

### CODE REVIEW

File: docs/02.API-REFERENCE.md [hunk 1]
```diff
@@ -53,6 +53,11 @@ Possible compile time enabled feature types:
  - JERRY_FEATURE_SNAPSHOT_EXEC - executing snapshot files
  - JERRY_FEATURE_DEBUGGER - debugging
  - JERRY_FEATURE_VM_EXEC_STOP - stopping ECMAScript execution
+ - JERRY_FEATURE_JSON - JSON functions
+ -  JERRY_FEATURE_PROMISE - promise builtin
```
Comment (LaszloLango) [inline docs/02.API-REFERENCE.md]: wrong indentation

File: docs/02.API-REFERENCE.md [hunk 2]
```diff
@@ -53,6 +53,11 @@ Possible compile time enabled feature types:
...
+ - JERRY_FEATURE_TYPEDARRAY - Typedarray builtin
+ -  JERRY_FEATURE_DATE - Date builtin
```
Comment (LaszloLango) [inline docs/02.API-REFERENCE.md]: wrong indentation

File: jerry-core/api/jerry.c [hunk 3]
```diff
@@ -865,6 +865,21 @@ bool jerry_is_feature_enabled (const jerry_feature_t feature)
...
+          || feature == JERRY_FEATURE_PROMISE
+#endif /* !CONFIG_DISABLE_ES2015_PROMISE_BUILTIN */
+#ifndef CONFIG_DISABLE_ES2015_TYPEDARRAY_BUILTIN
+          || feature == JERRY_FEATURE_TYPEDARRAY
+#endif /* !CONFIG_DISABLE_ES2015_TYPEDARRAY_BUILTIN */
+#ifndef CONFIG_DISABLE_DATE_BUILTIN
+          || feature == JERRY_FEATURE_DATE
+#endif /* !CONFIG_DISABLE_DATE_BUILTIN */
+#ifndef CONFIG_DISABLE_ES2015_TYPEDARRAY_BUILTIN
+          || feature == JERRY_FEATURE_REGEX
```
Comment (robertsipka) [inline jerry-core/api/jerry.c]: I think this is not JERRY_FEATURE_REGEX.

File: docs/02.API-REFERENCE.md [hunk 4]
```diff
@@ -53,6 +53,11 @@ Possible compile time enabled feature types:
  - JERRY_FEATURE_SNAPSHOT_EXEC - executing snapshot files
  - JERRY_FEATURE_DEBUGGER - debugging
  - JERRY_FEATURE_VM_EXEC_STOP - stopping ECMAScript execution
+ - JERRY_FEATURE_JSON - JSON functions
```
Comment (zherczeg) [inline docs/02.API-REFERENCE.md]: JSON support

File: docs/02.API-REFERENCE.md [hunk 5]
```diff
@@ -53,6 +53,11 @@ Possible compile time enabled feature types:
...
+ - JERRY_FEATURE_PROMISE - promise builtin
```
Comment (zherczeg) [inline docs/02.API-REFERENCE.md]: promise support
also change builtin -> support everywhere

File: jerry-core/include/jerryscript-core.h [hunk 6]
```diff
@@ -90,6 +90,11 @@ typedef enum
   JERRY_FEATURE_SNAPSHOT_EXEC, /**< executing snapshot files */
   JERRY_FEATURE_DEBUGGER, /**< debugging */
   JERRY_FEATURE_VM_EXEC_STOP, /**< stopping ECMAScript execution */
+  JERRY_FEATURE_JSON, /**< JSON functions */
+  JERRY_FEATURE_PROMISE, /**< promise builtin */
+  JERRY_FEATURE_TYPEDARRAY, /**< Typedarray builtin */
+  JERRY_FEATURE_DATE, /**< Date builtin */
+  JERRY_FEATURE_REGEX, /**< Regex builtin */
```
Comment (zherczeg) [inline jerry-core/include/jerryscript-core.h]: Update these.

**Comment by robertsipka** [comment]
LGTM (informally)

File: jerry-core/include/jerryscript-core.h [hunk 7]
```diff
@@ -90,6 +90,11 @@ typedef enum
   JERRY_FEATURE_SNAPSHOT_EXEC, /**< executing snapshot files */
   JERRY_FEATURE_DEBUGGER, /**< debugging */
   JERRY_FEATURE_VM_EXEC_STOP, /**< stopping ECMAScript execution */
+  JERRY_FEATURE_JSON, /**< JSON support */
+  JERRY_FEATURE_PROMISE, /**< promise support */
+  JERRY_FEATURE_TYPEDARRAY, /**< Typedarray support */
+  JERRY_FEATURE_DATE, /**< Date support */
+  JERRY_FEATURE_REGEX, /**< Regex support */
```
Comment (LaszloLango) [inline jerry-core/include/jerryscript-core.h]: The standard calls it `RegExp` and not `Regex`, so please rename to be consistent. http://www.ecma-international.org/ecma-262/5.1/index.html#sec-15.10
`JERRY_FEATURE_REGEXP, /**< RegExp support */`

File: docs/02.API-REFERENCE.md [hunk 8]
```diff
@@ -53,6 +53,11 @@ Possible compile time enabled feature types:
  - JERRY_FEATURE_SNAPSHOT_EXEC - executing snapshot files
  - JERRY_FEATURE_DEBUGGER - debugging
  - JERRY_FEATURE_VM_EXEC_STOP - stopping ECMAScript execution
+ - JERRY_FEATURE_JSON - JSON support
+ - JERRY_FEATURE_PROMISE - promise support
+ - JERRY_FEATURE_TYPEDARRAY - Typedarray support
+ - JERRY_FEATURE_DATE - Date support
+ - JERRY_FEATURE_REGEX - Regex support
```
Comment (LaszloLango) [inline docs/02.API-REFERENCE.md]: `JERRY_FEATURE_REGEXP - RegExp support`

File: jerry-core/api/jerry.c [hunk 9]
```diff
@@ -865,6 +865,21 @@ bool jerry_is_feature_enabled (const jerry_feature_t feature)
...
+          || feature == JERRY_FEATURE_PROMISE
+#endif /* !CONFIG_DISABLE_ES2015_PROMISE_BUILTIN */
+#ifndef CONFIG_DISABLE_ES2015_TYPEDARRAY_BUILTIN
+          || feature == JERRY_FEATURE_TYPEDARRAY
+#endif /* !CONFIG_DISABLE_ES2015_TYPEDARRAY_BUILTIN */
+#ifndef CONFIG_DISABLE_DATE_BUILTIN
+          || feature == JERRY_FEATURE_DATE
+#endif /* !CONFIG_DISABLE_DATE_BUILTIN */
+#ifndef CONFIG_DISABLE_REGEXP_BUILTIN
+          || feature == JERRY_FEATURE_REGEX
```
Comment (LaszloLango) [inline jerry-core/api/jerry.c]: `JERRY_FEATURE_REGEXP`

**Pull request merged**